wordpress-mobile/WordPress-Android
# [Blogging Prompts] Fetch Prompts List real data

Part of #17125 
Depends on FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2626

Update the `FetchBloggingPromptsListUseCase` to actually fetch data from FluxC, returning the correct list of prompts for today + the past 10 days (11 prompts in total).

A change is needed in FluxC to improve the `BloggingPromptsStore.fetchPrompts` usage (https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/2624). This PR updated the hash version of FluxC to point to the commit in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2626.

**Before merging this, we need to merge https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2626 and change this PR to point to a trunk hash version.**

Demo:

https://user-images.githubusercontent.com/5091503/210871628-7ef588ea-0ee9-49d2-afd4-dcbd2e1650c7.mp4

To test:
1. Do the steps to enable the Prompts List feature flag found in the test section of #17675
2. Sign in (if not signed in already)
3. Go to the home dashboard (My Site -> Home)
4. **Verify** the Prompts card is shown
5. Tap the 3-dot menu in the Prompts card
6. Tap the "View more prompts" option
7. **Verify** the Prompts List activity is shown
9. **Verify** the content is updated with the real list of past prompts (today + 10 days before) after loading

## Regression Notes
1. Potential unintended areas of impact
Blogging Prompts Card refresh/loading (in app home / my site).

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
Updated tests for `FetchBloggingPromptsListUseCase` and the ViewModel that uses it.

PR submission checklist:
